src/NumericalAlgorithms/Spectral/Spectral.cpp

```diff
@@ -369,6 +369,52 @@ Matrix interpolation_matrix(const size_t num_points,
   return interp_matrix;
 }
 
+const std::pair<Matrix, Matrix>& boundary_interpolation_matrix(
+    const Mesh<1>& mesh) noexcept {
+  ASSERT(mesh.basis(0) == Spectral::Basis::Legendre,
+         "We only support DG with a Legendre basis.");
+  ASSERT(mesh.quadrature(0) == Spectral::Quadrature::Gauss,
+         "We only compute the boundary interpolation for Gauss quadrature "
+         "since for Gauss-Lobatto you can just copy values off the volume.");
+  static const auto cache = make_static_cache<CacheRange<
+      1, Spectral::maximum_number_of_points<Spectral::Basis::Legendre>>>(
+      [](const size_t num_points) noexcept {
+        const Mesh<1> local_mesh{num_points, Spectral::Basis::Legendre,
+                                 Spectral::Quadrature::Gauss};
+        return std::pair<Matrix, Matrix>{interpolation_matrix(local_mesh, -1.0),
+                                         interpolation_matrix(local_mesh, 1.0)};
+      });
+  return cache(mesh.number_of_grid_points());
+}
+
+const std::pair<DataVector, DataVector>& boundary_lifting_term(
+    const Mesh<1>& mesh) noexcept {
+  ASSERT(mesh.basis(0) == Spectral::Basis::Legendre,
+         "We only support DG with a Legendre basis.");
+  ASSERT(mesh.quadrature(0) == Spectral::Quadrature::Gauss,
+         "We only compute the boundary interpolation for Gauss quadrature "
+         "since for Gauss-Lobatto you can just copy values off the volume.");
+  static const auto cache = make_static_cache<CacheRange<
+      1, Spectral::maximum_number_of_points<Spectral::Basis::Legendre>>>(
+      [](const size_t num_points) noexcept {
+        const Mesh<1> local_mesh{num_points, Spectral::Basis::Legendre,
+                                 Spectral::Quadrature::Gauss};
+        const auto& matrices = boundary_interpolation_matrix(local_mesh);
+        std::pair<DataVector, DataVector> result{
+            DataVector{local_mesh.number_of_grid_points()},
+            DataVector{local_mesh.number_of_grid_points()}};
+        for (size_t i = 0; i < local_mesh.number_of_grid_points(); ++i) {
+          result.first[i] = matrices.first(0, i);
+          result.second[i] = matrices.second(0, i);
```

Comment:
Should this really take the first row, not the last?

Reply:
There's only one row in the matrix, so I'm not sure what you're concerned about.